controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +123,265 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+				var storageClassClaimStorageType providerpb.FulfillStorageClassClaimRequest_StorageType
+				switch r.storageClassClaim.Spec.Type {
+				case "blockpool":
+					storageClassClaimStorageType = FulfillstorageclassclaimrequestBlockpool
+				case "sharedfilesystem":
+					storageClassClaimStorageType = FulfillstorageclassclaimrequestSharedfilesystem
+				}
+
+				// Call the `FulfillStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+				_, err = providerClient.FulfillStorageClassClaim(
+					r.ctx,
+					r.storageCluster.Status.ExternalStorage.ConsumerID,
+					r.storageClassClaim.Name,
+					r.storageClassClaim.Spec.EncryptionMethod,
+					storageClassClaimStorageType,
+				)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to initiate fulfillment of StorageClassClaim: %v", err)
+				}
+			}
+
+			// Update the StorageClassClaim status.
+			r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimConfiguring
+
+			// Call the `GetStorageClassClaimConfig` service on the provider server with StorageClassClaim as a request message.
+			response, err := providerClient.GetStorageClassClaimConfig(
+				r.ctx,
+				r.storageCluster.Status.ExternalStorage.ConsumerID,
+				r.storageClassClaim.Name,
+			)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to get StorageClassClaim config: %v", err)
+			}
+
+			resources := response.ExternalResource
+			if resources == nil {
+				return reconcile.Result{}, fmt.Errorf("no configuration data recieved")
+			}
+
+			// Go over the received objects and operate on them accordingly.
+			for _, resourceItem := range resources {
+				data := map[string]string{}
+				err = json.Unmarshal(resourceItem.Data, &data)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to unmarshal StorageClassClaim configuration response: %v", err)
+				}
+				resource := storagecluster.ExternalResource{
+					Name: resourceItem.Name,
+					Data: data,
+					Kind: resourceItem.Kind,
+				}
+
+				switch resource.Kind {
+				case "Secret":
+					secret := &corev1.Secret{}
+					secret.Name = resource.Name
+					secret.Namespace = r.storageClassClaim.Namespace
+					secret.Data = map[string][]byte{}
+					for key, value := range resource.Data {
+						secret.Data[key] = []byte(value)
+					}
+					err := r.own(secret)
+					if err != nil {
+						return reconcile.Result{}, fmt.Errorf("failed to own Secret: %v", err)
+					}
+					_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, secret, func() error {
+						return nil
+					})
+					if err != nil {
+						return reconcile.Result{}, fmt.Errorf("failed to create or update secret %v: %s", secret, err)
+					}
+				case "StorageClass":
+					var storageClass *storagev1.StorageClass
+					if resource.Name == "cephfs" {
+						err = r.rookEnableCephFSCSI()
+						if err != nil {
+							return reconcile.Result{}, fmt.Errorf("failed to enable cephfs csi: %v", err)
+						}
```

Comment:
This does not make sense here as it should be done only once per the entire cluster and not for every claim!
Also, the entire operation has nothing to do with the current claim and does not use any of the claim data.
Let's move this into the storagecluster controller and make sure to enable CephCSI every time we are in consumer mode (regardless of claims)